feat: system framework hooks

xfqwdsj opened the pull request:
Closes: #104 
Closes: #26 
Fixes: #4 
Fixes: #9 

## 注意 Notes

本PR仅对<https://cs.android.com/android/_/android/platform/frameworks/base/+/1109a0c5446c311c2fb2c97b5ee6253cd624e0e3>之后的提交保证兼容，对于更早的版本，我编写了基本的向后兼容支持，但不保证可用。欢迎PR。

This PR guarantees compatibility only with commits after <https://cs.android.com/android/_/android/platform/frameworks/base/+/1109a0c5446c311c2fb2c97b5ee6253cd624e0e3>. For earlier versions, I've written basic backward compatibility support, but its usability is not guaranteed. PRs are welcome.

## 状态 Updates

由于各种原因，该项开发暂时搁置。

Postponed.

xfqwdsj commented:
@igormiguell would you help me to translate the strings?

igormiguell commented:
> @igormiguell would you help me to translate the strings?

of course

xfqwdsj committed:
refactor: change entrypoint

**WARNING**: READ BEFORE TRY

**Known Issues**

- 开关完全不可用
- adb无法连接

xfqwdsj commented:
good

![good](https://github.com/user-attachments/assets/a909572d-938b-4e37-ab6c-78c48e2f302e)

xfqwdsj commented:
<details><summary>Screenshots</summary>

<img width="3840" height="2064" alt="image" src="https://github.com/user-attachments/assets/c6ce817d-0773-4bfb-a3cd-f141a331892a" />

<img width="3840" height="2064" alt="image" src="https://github.com/user-attachments/assets/a0727922-f24a-45c9-9b00-6785f0bcb216" />

</details>

xfqwdsj committed:
feat: settings provider hook

**Known Issues**

- 所有功能不会工作，仅可在应用内测试

xfqwdsj committed:
refactor: 使用广播触发更新

xfqwdsj committed:
refactor: 加回properties（无法在系统框架内实现）

xfqwdsj committed:
refactor: 更改结构，优化r8

xfqwdsj committed:
refactor: 改名

xfqwdsj committed:
chore: 添加导航

xfqwdsj committed:
feat: aidl binder (this commit not works)

xfqwdsj committed:
refactor

xfqwdsj committed:
refactor: 直接通过自定义的SettingsProvider事务通知更新

xfqwdsj committed:
feat: 注入ContentProvider

xfqwdsj committed:
refactor: 清理其他代码

xfqwdsj committed:
refactor: 使用dsl以增加灵活性

xfqwdsj committed:
fix: 条件写反

xfqwdsj committed:
feat: 使用导航

xfqwdsj committed:
feat: 迁移导航

xfqwdsj committed:
feat: 更好的页面

xfqwdsj committed:
feat: compatibility (?) use with care

PRs welcome

xfqwdsj committed:
fix: hook错误

xfqwdsj committed:
refactor: 重命名检测类

xfqwdsj committed:
feat: 数据库（正在重构，该commit无法工作）

xfqwdsj committed:
refactor: 更改provider注入点

xfqwdsj committed:
chore: 正在重构

xfqwdsj committed:
fix: 服务错误

xfqwdsj committed:
refactor: 优化hook

xfqwdsj committed:
chore: 正在编写数据库读取

xfqwdsj committed:
feat: 数据库

xfqwdsj committed:
refactor: make ParcelablePackageInfo fields non-nullable

This commit changes the fields `packageName`, `userId`, and `appId` in the `ParcelablePackageInfo` data class to be non-nullable.

xfqwdsj committed:
refactor: service

Remove unused `connections` property and `DaoConnectionsMap` class from `NotDevService`.

xfqwdsj committed:
refactor: clear calling identity in notifySettingChange

This change calls `clearCallingIdentity` within the `notifySettingChange` method in `NotDevService.kt`.

xfqwdsj committed:
refactor: 移除aidl，改进日志处理

xfqwdsj committed:
chore: add icons

xfqwdsj committed:
build: drop aidl

xfqwdsj committed:
feat: GroupedLazyList

xfqwdsj committed:
refactor: pass service instance to SystemServiceBuilder blocks

This change modifies the `SystemServiceBuilder` to pass the `SystemServiceInterface` instance to the `queryApps` and `notifySettingChange` lambda blocks. This allows these blocks to call other methods on the service if needed.

xfqwdsj committed:
refactor: allow nullable log message

xfqwdsj committed:
refactor: 消除硬编码

xfqwdsj committed:
feat: 实现应用查询

xfqwdsj committed:
refactor: 停止使用kaidl的suspend transaction

因为其会导致binder调用者信息丢失

xfqwdsj committed:
refactor: 提取card color

xfqwdsj committed:
chore: 加入adaptive

xfqwdsj committed:
fix: 不符合预期的导航行为

xfqwdsj committed:
feat: 加入页面并调整导航

xfqwdsj committed:
chore: 由于加入了扩展图标包，修改图标

xfqwdsj committed:
refactor: 提取ListItemColors

xfqwdsj committed:
refactor: 重命名data包到database，删除冗余hook

xfqwdsj committed:
feat: 处理数据库操作异常

xfqwdsj committed:
feat: SystemServiceClient

xfqwdsj committed:
feat: 用户信息查询

xfqwdsj committed:
chore: add coil

xfqwdsj commented:
<img width="3840" height="2064" alt="image" src="https://github.com/user-attachments/assets/7e8da98a-def0-46d8-873e-4b3608c0f985" />

xfqwdsj committed:
chore: 提出PaddingValues实用属性

xfqwdsj committed:
feat: GroupedLazyList

xfqwdsj committed:
chore: add haze

xfqwdsj committed:
refactor: service

xfqwdsj committed:
feat: 更新Padding实用函数

xfqwdsj committed:
feat: 更新WindowInsets实用函数

xfqwdsj committed:
refactor: Color

xfqwdsj committed:
feat: Haze实用函数

xfqwdsj committed:
refactor: 更新ui结构

xfqwdsj committed:
chore: 移除测试代码

xfqwdsj committed:
refactor: 查询package info

xfqwdsj committed:
refactor: 数据库服务绑定

xfqwdsj committed:
feat: implement NotDevApplication for database initialization

xfqwdsj committed:
refactor: simplify NotDevApplication initialization and update BinderProvider usage

xfqwdsj committed:
fix: initialization issues

xfqwdsj committed:
feat: add PaddingValues extension for WindowInsets conversion

xfqwdsj committed:
chore: remove unused imports

xfqwdsj committed:
feat: add custom easing for HazeEffect

xfqwdsj committed:
fix: lazy list

xfqwdsj committed:
chore: remove unused imports

xfqwdsj committed:
chore: 继续调整haze参数

xfqwdsj committed:
refactor: view model

xfqwdsj committed:
chore: 新增字符串资源

xfqwdsj committed:
feat: add IconButtonWithTooltip composable

xfqwdsj committed:
chore: update string resource names

xfqwdsj committed:
feat: integrate IconButtonWithTooltip for improved action buttons

xfqwdsj committed:
feat: 可自定义提示框位置的 IconButton

xfqwdsj committed:
refactor: 移动IconButon

放错了...

xfqwdsj committed:
chore: reorder strings

xfqwdsj committed:
feat: filter dialog

xfqwdsj committed:
chore: update padding for bottom sheet app filter labels

xfqwdsj committed:
fix: HazeSnackbar

xfqwdsj committed:
chore: upgrade haze

xfqwdsj committed:
chore: upgrade navigation

xfqwdsj committed:
chore: upgrade compose

xfqwdsj committed:
feat: implement header for card lazy group

xfqwdsj committed:
refactor: 迁移CategoryCard到GroupedLazyColumn

xfqwdsj committed:
refactor: GroupedLazyList支持item

xfqwdsj committed:
chore: strings

xfqwdsj committed:
feat: user filter

xfqwdsj committed:
chore: add easing to Haze

xfqwdsj committed:
feat: Alpha Mask 实用函数

xfqwdsj committed:
fix: 阻止LazyRow响应指定区域的点击事件

xfqwdsj committed:
refactor: 数据库代码整合

xfqwdsj committed:
refactor: 简化数据库初始化

xfqwdsj committed:
refactor: 多项ui与架构更新

xfqwdsj committed:
chore: add datastore

xfqwdsj committed:
chore: add kotlinx serialization cbor

xfqwdsj committed:
chore: upgrade kotlin

xfqwdsj commented:
App list

<details>

<img width="1344" height="2992" alt="image" src="https://github.com/user-attachments/assets/34b7a84d-1f62-4b5a-997e-faedd7621c12" />

</details>

xfqwdsj committed:
feat: 筛选

xfqwdsj committed:
feat: 应用列表为空时增加提示

xfqwdsj committed:
refactor: AppListStatus调整

AppListStatus的isAllFiltered调整为isFiltered，并更新相关逻辑

xfqwdsj commented:
Filter

<details><summary>All</summary>

<img width="1344" height="2992" alt="image" src="https://github.com/user-attachments/assets/7928ae4b-b04b-46f2-aa7d-e2a3df04a34b" />

</details>

<details><summary>Only user</summary>

<img width="1344" height="2992" alt="image" src="https://github.com/user-attachments/assets/80aa21b3-4a94-4f74-89f3-970fa0de7445" />

</details>

xfqwdsj committed:
refactor: 调整CREATOR

移除JvmStatic，移除类型注解

xfqwdsj committed:
chore: 优化import

xfqwdsj committed:
feat: 添加模糊效果的动画可见性组件

xfqwdsj committed:
chore: 优化import

xfqwdsj committed:
chore: 更新adaptive

xfqwdsj committed:
chore: 整理Apps代码

xfqwdsj committed:
chore: 更改animation label命名

xfqwdsj committed:
refactor: 重新组织Apps

提取组件

xfqwdsj committed:
refactor: AppListItem新增modifier

xfqwdsj committed:
refactor: haze

xfqwdsj committed:
chore: 微调AppConfiguration

xfqwdsj committed:
chore: 微调App

xfqwdsj committed:
chore: 移除消耗insets

xfqwdsj committed:
chore: 提取出AnimatedContentDefaultTransform

xfqwdsj committed:
refactor: 更新AnimatedVisibilityWithBlur动画

xfqwdsj committed:
chore: 优化import

xfqwdsj committed:
chore: Haze允许传入zIndexDelta

xfqwdsj committed:
chore: strings

xfqwdsj committed:
feat: splashscreen

xfqwdsj committed:
refactor: 移动状态位置

xfqwdsj committed:
chore: 重命名字符串，更清晰

xfqwdsj committed:
refactor: rename appList to appListCard and enhance semantics

xfqwdsj committed:
fix: 内容颜色

xfqwdsj committed:
fix: update app list processing to use queried configured list

xfqwdsj committed:
chore: 调整padding

xfqwdsj committed:
chore: strings

xfqwdsj committed:
refactor: 将配置框状态移动到viewmodel

xfqwdsj committed:
fix: 布局跳动

xfqwdsj committed:
fix: 动画错误

xfqwdsj committed:
feat: 在对话框中处理返回

由于[bug](https://issuetracker.google.com/issues/439262603)，导航回退到了1.0.0-alpha06

xfqwdsj committed:
chore: 缩小对话框边距

xfqwdsj committed:
chore: 把提供接收器换成提供上下文

xfqwdsj committed:
chore: 筛选掉无applicationInfo的包（尽管它通常不会发生）

xfqwdsj committed:
refactor: 用context替换receiver

xfqwdsj committed:
feat: 反射获得uid相关方法

xfqwdsj committed:
chore: 修改变量名称

xfqwdsj committed:
chore: 修改命名

xfqwdsj committed:
feat: 通过自身的packageInfo获取userId

xfqwdsj committed:
feat: AppConfiguration添加数据库初始化操作

xfqwdsj committed:
feat: 获取PackageInfo

xfqwdsj committed:
feat: 应用列表header添加动画

xfqwdsj committed:
refactor: 应用列表计算

同时修复了在应用配置状态切换时的闪退

xfqwdsj committed:
chore: use context instead of with

xfqwdsj committed:
fix: a weird ui bug

...maybe?

xfqwdsj committed:
chore: 拆分remember...会更优吗？

xfqwdsj committed:
chore: 移除激进优化

xfqwdsj committed:
refactor: 更改uid相关函数

xfqwdsj committed:
feat: add AnimatedVisibilityWithBlur composable for enhanced visibility transitions

xfqwdsj committed:
fix: 筛选导致的问题

xfqwdsj committed:
refactor: 使用自定义实现标志package info唯一性

xfqwdsj committed:
fix: 给NoAppsBackground传入正确的padding

xfqwdsj committed:
feat: 加强BlurAnimation

xfqwdsj committed:
feat: lookahead实用函数

xfqwdsj committed:
refactor: 筛选弹窗

xfqwdsj committed:
chore: 模糊支持BlurredEdgeTreatment参数

xfqwdsj committed:
refactor: 优化状态卡片布局和无障碍

xfqwdsj committed:
chore: upgrade agp

xfqwdsj committed:
feat: 增强无障碍

xfqwdsj committed:
chore: 修改datastore文件名

xfqwdsj committed:
fix: card group在组中只有一项时形状计算错误

xfqwdsj committed:
feat: 新增全局设置项目

xfqwdsj committed:
refactor: 将原PreferenceItem重命名为DetectionItem，构造新的通用PreferenceItem

xfqwdsj committed:
feat: 在概览页面新增全局设置开关

xfqwdsj committed:
chore: 更改用户id源

xfqwdsj committed:
refactor: 将关闭逻辑移出

xfqwdsj committed:
feat: 优化用户选择无障碍

xfqwdsj committed:
feat: 优化列表无障碍

xfqwdsj committed:
feat: 优化应用列表无障碍

xfqwdsj committed:
feat: 优化状态卡片无障碍

xfqwdsj committed:
chore: 文字调整为图标

xfqwdsj committed:
refactor: Color移动到theme

xfqwdsj committed:
chore: 调整形状

xfqwdsj committed:
chore: rename

xfqwdsj committed:
fix: snackbar显示位置错误

xfqwdsj commented:
@nrmu9 Working in progress. Functions that already implemented (not including dev hiding) are tested on Android 16. If you want to try (and test), the latest build is ok.

xfqwdsj committed:
refactor: 用flow重构应用列表查询

同时实现了列表下拉刷新，减少了切换到应用列表时提示框对用户的打扰

hexman1999 commented:
@xfqwdsj 

I know work is in progress, but I suggest something I noticed, 


Mostly system apps don't require hiding, so I suggest to untick it from filters by default or use checkbox instead which is unchecked by default, if you need to view, tick



Other thing that user apps not showing maybe not yet implemented but just telling what I noticed

Thank you!

xfqwdsj committed:
chore: filter out system apps by default

xfqwdsj committed:
fix: values are not updated from datastore

hexman1999 commented:
@xfqwdsj 

Noticed that blur making low end devices lag in loading and scrolling, maybe an option can be added for enable or disable that to make app smooth,

You are spending lot of time on developing this really needed application, I am following the Pull request everyday to test latest version, I appreciate that, thank you, I wish you the best

xfqwdsj commented:
@hexman1999 

Yes, I will add some switches to toggle visual effects - that's indeed needed.

Default value of filters is updated. Tou can clear data to see the changes.

Feel free to share any ideas!

xfqwdsj committed:
chore: adjust rotation animation

xfqwdsj committed:
refactor: rework of shape system

- introduce g2 corner
- refactor GroupedLazyList shape calculation

xfqwdsj committed:
fix: haze计算错误

xfqwdsj committed:
feat: fab menu in apps page

also:

- extract EmptyContentTransform
- extract FocusRequestingEffect
- move snackbar from app to page for a better ui
- implement HazeFloatingActionButton
- adjust HazeSnackbar to adapt the behavior changes of snackbar
- add strings

xfqwdsj committed:
feat: clear binder identity when requesting app list

xfqwdsj committed:
refactor: extract FilterChip

hexman1999 commented:
Is it better to use system applist query instead of lsposed?, I think most apps using it to get applist, including databackup and even magisk


Noticed that when used hyperos which has an option in permissions that is disabled by default which is permission to access applist, i had to allow before using either apps

Here I have to enable lsposed to get the list,

-------------------
(Just wondering 🤔, can be there any benefit made if the app used root permission in anything?, I noticed that HideMyApplist using root permission for certain functions to toggle options for better hiding its traces)

xfqwdsj committed:
fix: missing spacing in app list

xfqwdsj committed:
fix: wrong animation in app list

xfqwdsj committed:
refactor: app lists collecting

hexman1999 commented:
initially crash, after reboot it opened, not loading apps, with or without the permission with warning triangle in applist that says failed to querey apps, note that pressing any button can work once or twice then crash


I am using HOS 1.0.20 Custom rom on Redmi Note 10 Pro, it can be a problem with the port as my phone didn't receive it officially, but i am reporting what I see

xfqwdsj commented:
Yes there is a bug querying apps when every time opening the app and you can pull to refresh the app list. Is the crash happen continuously? Please catch the logs after crash, thank you.

xfqwdsj committed:
refactor: Flow操作

- 修复了大量应用列表相关问题
- 给Main page增加了secondClick函数
- 二次点击应用列表导航菜单即可刷新应用列表

xfqwdsj commented:
I found some issues. I'll mention it after I resolve them.

xfqwdsj committed:
refactor: rename database clearAllData to clearAllPackages

xfqwdsj committed:
chore: upgrade agp

xfqwdsj committed:
refactor: 全局偏好使用数据库

xfqwdsj committed:
fix: 条件写反

同时解开了概览页开关的使用限制，移除了不使用的属性

xfqwdsj committed:
chore: upgrade agp
